core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -371,6 +377,16 @@ case class DefBlackBox(
   params: Map[String, Param])
     extends Component
 
+@nowarn("msg=class Port") // delete when Port becomes private
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
+case class DefIntrinsicModule(
```

Comment:
The deprecation can be avoided here by going directly to making this private:
```suggestion
private[chisel3] case class DefIntrinsicModule(
```